webui/react/src/components/kit/Confirmation.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { PropsWithChildren, ReactNode, useState } from 'react';
+
+import { Modal, useModal } from './Modal';
+
+export interface ConfirmModalProps {
+  cancelText?: string;
+  content?: ReactNode;
+  danger?: boolean;
+  title?: string;
+  okText?: string;
+  onClose?: () => void;
+  onConfirm?: () => Promise<void>;
+}
+
+export const DEFAULT_CONFIRM_TITLE = 'Confirm Action';
+export const DEFAULT_CONFIRM_LABEL = 'Confirm';
+export const DEFAULT_CONTENT = 'Are you sure?';
+
+const ConfirmModal = ({
+  cancelText,
+  content,
+  danger = false,
+  title,
+  okText,
+  onClose,
+  onConfirm,
+}: ConfirmModalProps) => {
+  return (
+    <Modal
+      cancel
+      cancelText={cancelText}
+      danger={danger}
+      icon="warning-large"
+      size="small"
+      submit={{
+        handler: onConfirm ?? voidPromiseFn,
+        text: okText ?? DEFAULT_CONFIRM_LABEL,
+      }}
+      title={title ?? DEFAULT_CONFIRM_TITLE}
+      onClose={onClose}>
+      <div>{content}</div>
+    </Modal>
+  );
+};
+
+type VoidFn = () => void;
+type VoidPromiseFn = () => Promise<void>;
+type ConfirmModalModifier = (args: ConfirmModalProps) => void;
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+const voidFn = () => {};
+const voidPromiseFn = async () => {};
+const ConfirmationContext = React.createContext<ConfirmModalModifier | null>(null);
+
+export const ConfirmationProvider: React.FC<PropsWithChildren> = ({ children }) => {
+  const [cancelText, setCancelText] = useState<string>();
+  const [content, setContent] = useState<ReactNode>();
+  const [danger, setDanger] = useState(false);
+  const [okText, setOkText] = useState<string>();
+  const [title, setTitle] = useState<string>();
+  const [onClose, setOnClose] = useState<VoidFn>(voidFn);
+  const [onConfirm, setOnConfirm] = useState<VoidPromiseFn>(voidPromiseFn);
```

Comment:
option: since you're initializing and setting all of these in the same place it might make sense to put them all in the same state or use a reducer. Only if it's not too much hassle

Reply:
great suggestion. I ended up using a consolidated useState, since they were all getting set at the same time.